valexa/ploting/ploters.py

```diff
@@ -0,0 +1,42 @@
+import json
+from typing import Dict, List
+
+from valexa.core.profiles import (DEFAULT_ACCEPTANCE, DEFAULT_TOLERANCE,
+    Profile, make_profiles)
+from valexa.core.xlsx import XlsxHandler
+from valexa.ploting.plotly.canvas import ProfilePlotCanvas
+
+
+class PloterData:
+
+    handler = XlsxHandler
+    calib_data = None
+    valid_data = None
+    profiles = []
+    tolerance_limit = DEFAULT_TOLERANCE
+    acceptance_limit = DEFAULT_ACCEPTANCE
+
+    def __init__(self, file: str or File=None, **kwargs):
+        if file:
+            self.load_data(file)
+
+        for key, value in kwargs.items():
```

Comment:
what's happen here? I'm not sure to understand correctly. It would be a good idea to extract this chunk in a well-named function. However, can you explain the purpose if this?

Reply:
The loop is for using composition instead of inheritance. For example if all you need is a different handler or limits, you just need to pass it as a keyword argument, rather than having to make a new subclass:
`data = PloterData(handler=MyHandlerClass)`
You only need to create a subclass if you need to modify the class' behavior. 

(Note: the loop was supposed to be before the data loading and has been corrected)

Reply:
Why you did not use a simple `PloterData(handler=MyHandlerClass)` rather than use kwargs?

Reply:
I could explicitly check the kwargs dict rather than looping through it:
```
if "handler" in kwargs:
   self.handler = kwargs["handler"]
if "tolerance_limit" in kwargs:
    self.handler = kwargs["tolerance_limit"]
if "acceptance_limit" in kwargs:
    self.handler = kwargs["acceptance_limit"]
```
I prefer not to use optional positional arguments in __init__ methods because it becomes the responsibility of the subclasses to take and pass them on to the parent. They'll have to either add them to their signature on top of their own optional args, or one of them is going to use kwargs eventually.

Reply:
to be honest I'm lost. Can you find me some example that could prove your point, please?

Reply:
Never mind, I forgot. that this was possible:
```
class MyClass:
    def __init__(self, test="no"):
        self.test = test

class PossibleFutureSubclass(MyClass):
    def __init__(self, something, **myclass_kwargs):
        self.something = something
        super().__init__(**myclass_kwargs) #will populate optional args

b = PossibleFutureSubclass("something")
# b.test == "no"
b = PossibleFutureSubclass("something", test="yes")
# b.test == "yes"
```

So MyClass can have optional positional args and a future subclass doesn't have to manage them much. I'll change for optional arguments.